packages/expo-image/src/ExpoImage.web.tsx

```diff
@@ -10,14 +10,18 @@ import useSourceSelection from './web/useSourceSelection';
 loadStyle();
 
 export const ExpoImageModule = {
-  prefetch(urls: string | string[]): void {
+  prefetch(urls: string | string[]): Promise<boolean[]> {
     const urlsArray = Array.isArray(urls) ? urls : [urls];
-    urlsArray.forEach((url) => {
-      const img = new Image();
-      img.src = url;
+    const promises = urlsArray.map((url) => {
```

Comment:
Since we're using Promise.all, if we could reject the promise on img load error, it would immediately reject the main promise and we wouldn't need to wait for all images to load before returning a failure state. We would need to convert the rejection to a resolution with `false` returned somehow.